js/src/setup-mc/setup-stepper/setup-free-listings/index.js

```diff
@@ -59,8 +70,40 @@ const SetupFreeListings = ( props ) => {
 		return {};
 	};
 
-	const handleSubmitCallback = () => {
-		onContinue();
+	const handleSubmitCallback = async ( values ) => {
+		/**
+		 * Even though we already have auto-save effects in the FormContent,
+		 * we are saving the form values here again to be sure,
+		 * because the auto-save may not be fired
+		 * when users were having focus on the text input fields
+		 * and then immediately click on the Continue button.
+		 */
```

Comment:
💅 
This detail is very thoughtful, but I doubt merchants would easily get stuck in this situation and cannot continue. Even the free listings edit page is always like this from the beginning. Adding this code as an exchange for that processing seems a bit unworthy.

Reply:
> Even the free listings edit page is always like _this_ from the beginning.

Sorry, what do you mean by "this"? I don't get it. 😅 

In Edit Free Listings, we click on the "Save changes" button, the button will show a loading icon, and we do the saving in the background.

In this Setup MC here, I think it is good to save upon users' action on clicking the button. Even though we already have the "auto-save" feature, I'd say "auto-save" is a "nice-to-have" feature, it can be removed anytime if we want, and it may fail silently when it fails. On the other hand, "save upon button click" is a "must-have" feature, and when it fails, we must inform users and we cannot proceed to the next step.

Reply:
> what do you mean by "this"? I don't get it.

Sorry, I misunderstood it as another UI detail that is not a problem actually. Please ignore that comment. 🙏 

---

> On the other hand, "save upon button click" is a "must-have" feature, and when it fails, we must inform users and we cannot proceed to the next step.

If this is a concern, then the shipping time seems to be missed in this handling?